include/boost/numeric/ublas/tensor/tensor_core.hpp

```diff
@@ -0,0 +1,786 @@
+     *
+     * @code tensor_core<float> A{extents{4,2,3}}; @endcode
+     *
+     * @param s initial tensor_core dimension extents
+     */
+    template<typename U = extents_type,
+        typename = std::enable_if_t<is_dynamic_v<U>>
+    >
+    explicit inline
+    tensor_core (extents_type const& s)
+        : tensor_core( s, resizable_tag{} )
+    {}
+
+    /** @brief Constructs a tensor_core with a \c shape
+     *
+     * By default, its elements are initialized to 0.
+     *
+     * @code tensor_core<float> A{extents{4,2,3}}; @endcode
+     *
+     * @param s initial tensor_core dimension extents
+     * @param i initial tensor_core with this value
+     */
+    template<typename U = extents_type,
+        typename = std::enable_if_t<is_dynamic_v<U>>
+    >
+    explicit inline
+    tensor_core (extents_type const& s, value_type const& i)
+        : tensor_core( s, resizable_tag{} )
+    {
+        std::fill_n(begin(),size(),i);
+    }
+
+    /** @brief Constructs a tensor_core with a \c shape
+     *
+     * By default, its elements are initialized to 0.
+     *
+     * @code tensor_core<float> A{}; @endcode
+     *
+     * @param i initial tensor_core with this value
+     */
+    template<typename U = extents_type,
+        typename = std::enable_if_t<is_static_v<U>>
+    >
+    explicit inline
+    tensor_core (value_type const& i)
+        : tensor_core()
+    {
+        std::fill_n(begin(),size(),i);
+    }
+
+    /** @brief Constructs a tensor_core with a \c shape and initiates it with one-dimensional data
+     *
+     * @code tensor_core<float> A{extents{4,2,3}, array }; @endcode
+     *
+     *
+     *  @param s initial tensor_core dimension extents
+     *  @param a container of \c array_type that is copied according to the storage layout
+     */
+    template<typename U = extents_type,
+        typename = std::enable_if_t<is_dynamic_v<U>>
+    >
+    inline
+    tensor_core (extents_type const& s, const array_type &a)
+        : tensor_core( s, resizable_tag{} )
+    {
+        if( size() != a.size() ){
+            throw std::runtime_error("boost::numeric::ublas::tensor_core(extents_type,array_type): "
+                "array size mismatch with extents"
+            );
+        }
+        std::copy_n(a.begin(),size(),begin());
+    }
+
+    /** @brief Constructs a tensor_core with a \c shape and initiates it with one-dimensional data
+     *
+     * @code tensor_core<float> A{ array }; @endcode
+     *
+     *  @param a container of \c array_type that is copied according to the storage layout
+     */
+    template<typename U = extents_type,
+        typename = std::enable_if_t<is_static_v<U>>
+    >
+    inline
+    tensor_core (const array_type &a)
+        : tensor_core()
+    {
+        if( size() != a.size() ){
+            throw std::runtime_error("boost::numeric::ublas::tensor_core(extents_type,array_type): "
+                "array size mismatch with extents"
+            );
```

Comment:
```suggestion
            throw std::runtime_error("boost::numeric::ublas::tensor_core(extents_type,array_type): "
                "array size mismatch with extents");
```

Readability.

Reply:
'clang format' will restructure it. So it's useless it format it.